assets/src/edit-story/components/library/panes/pageTemplates/templateSave.js

```diff
@@ -80,38 +99,79 @@ const StyledText = styled(Text)`
   margin-bottom: 8px;
 `;
 
-const StyledButton = styled(Button)`
-  height: 32px;
-`;
+function TemplateSave({ pageSize, setShowDefaultTemplates, loadTemplates }) {
+  const {
+    actions: { addPageTemplate },
+  } = useAPI();
+  const { showSnackbar } = useSnackbar();
+
+  const { currentPage } = useStory(({ state: { currentPage } }) => ({
+    currentPage,
+  }));
 
-function TemplateSave({ pageSize }) {
   const { customPageTemplates } = useFeatures();
-  const handleSaveTemplate = useCallback(() => {}, []);
+  const handleSaveTemplate = useCallback(() => {
+    // Don't add empty page.
+    if (isDefaultPage(currentPage)) {
+      showSnackbar({
+        message: __(
+          'An empty page can’t be saved as a template. Add elements and try again.',
+          'web-stories'
+        ),
+        dismissable: true,
+      });
+    } else {
+      addPageTemplate({ ...currentPage, id: uuidv4(), title: null }).then(
```

Comment:
Why are id and title being passed here? What do they do?

Reply:
This is basically just not to save templates with duplicate ID-s. Otherwise, it would take the page's ID and if the same page is used twice for creating a template, two templates would have the same ID.

Title is being removed just in case the page template is created from a page that had a title from the origin template.